pkg/services/accesscontrol/evaluator/evaluator.go

```diff
@@ -5,13 +5,18 @@ import (
 
 	"github.com/gobwas/glob"
 
+	"github.com/grafana/grafana/pkg/infra/metrics"
 	"github.com/grafana/grafana/pkg/models"
 	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // Evaluate evaluates access to the given resource, using provided AccessControl instance.
 // Scopes are evaluated with an `OR` relationship.
 func Evaluate(ctx context.Context, ac accesscontrol.AccessControl, user *models.SignedInUser, action string, scope ...string) (bool, error) {
+	timer := prometheus.NewTimer(metrics.MAccessPermissionsSummary)
```

Comment:
Won't this be registered twice per operation if we do it both in Evaluate and GetUserPermissions?

Reply:
I think this use is ok.  NewTimer doesn't register a metric, it does the measurement and writes it to metrics.MAccessPermissionsSummary, which is registered elsewhere.  timer is local to this call and is destroyed after every call, MAccessPermissionsSummary is the persistent metric. 

I hope.